modules/gapi/perf/streaming/gapi_streaming_source_perf_tests.cpp

```diff
@@ -82,18 +89,216 @@ PERF_TEST_P_(VideoCapSourcePerfTest, TestPerformance)
+                               source_description_preproc_t(files[0], codec[0], "", cif),
+                               source_description_preproc_t(files[0], codec[0], "MFX_ACCEL_MODE_VIA_D3D11", full_hd),
+                               source_description_preproc_t(files[0], codec[0], "MFX_ACCEL_MODE_VIA_D3D11", cif),
+                               source_description_preproc_t(files[1], codec[1], "", full_hd),
+                               source_description_preproc_t(files[1], codec[1], "", cif),
+                               source_description_preproc_t(files[1], codec[1], "MFX_ACCEL_MODE_VIA_D3D11",full_hd),
+                               source_description_preproc_t(files[1], codec[1], "MFX_ACCEL_MODE_VIA_D3D11",cif),
+                               source_description_preproc_t(files[2], codec[2], "", full_hd),
+                               source_description_preproc_t(files[2], codec[2], "", cif),
+                               source_description_preproc_t(files[2], codec[2], "MFX_ACCEL_MODE_VIA_D3D11", full_hd),
+                               source_description_preproc_t(files[2], codec[2], "MFX_ACCEL_MODE_VIA_D3D11", cif)));
+
+class OneVPLSourcePerf_PP_Engine_Test : public TestPerfParams<source_description_preproc_t> {};
+cv::GFrameDesc mock_network_info(int width, int height) {
+    cv::GFrameDesc ret {cv::MediaFormat::NV12,
+                        {width, height}};
+    return ret;
+}
+
+PERF_TEST_P_(OneVPLSourcePerf_PP_Engine_Test, TestPerformance)
+{
+    using namespace cv::gapi::wip;
+    using namespace cv::gapi::wip::onevpl;
+
+    const auto params = GetParam();
+    source_t src = findDataFile(get<0>(params));
+    codec_t type = get<1>(params);
+    accel_mode_t mode = get<2>(params);
+    resize_t res = get<3>(params);
+
+    std::vector<CfgParam> cfg_params {
+        CfgParam::create_implementation("MFX_IMPL_TYPE_HARDWARE"),
+    };
+
+    if (!type.empty()) {
+        cfg_params.push_back(CfgParam::create_decoder_id(type.c_str()));
+    }
+
+    if (!mode.empty()) {
+        cfg_params.push_back(CfgParam::create_acceleration_mode(mode.c_str()));
+    }
+
+    auto device_selector = std::make_shared<CfgParamDeviceSelector>(cfg_params);
+    auto source_ptr = cv::gapi::wip::make_onevpl_src(src, cfg_params, device_selector);
+
+    // create VPP preproc engine
+    auto required_frame_param = mock_network_info(res.first, res.second);
+    std::unique_ptr<VPLAccelerationPolicy> policy;
+    if (mode == "MFX_ACCEL_MODE_VIA_D3D11") {
+        policy.reset(new VPLDX11AccelerationPolicy(device_selector));
+    } else {
+        policy.reset(new VPLCPUAccelerationPolicy(device_selector));
+    }
+    VPPPreprocEngine preproc_engine(std::move(policy));
+    cv::gapi::wip::Data out;
+    TEST_CYCLE()
+    {
+        source_ptr->pull(out);
+        cv::MediaFrame frame = cv::util::get<cv::MediaFrame>(out);
+        cv::util::optional<pp_params> param = preproc_engine.is_applicable(frame);
```

Comment:
Since all of these methods must be called in order, for me it's simple to see the single function:
```
preproc_engine->apply(frame, out_frame_desc);
```

Where `frame` is input and `out_frame_desc` describe which frame user want to get as the result of preprocessing.

Reply:
these methods called in order in order to measure total execution cost. In real usage there may be another bunch of methods in execution flow in between `is_applicable` an `run_sync`
On the other side if `user` or another `developer` would see necessity in additional method `apply` - they could implement it easily as an aggregation two of these methods